platforms/android/tangram/src/main/java/com/mapzen/tangram/DefaultHttpHandler.java

```diff
@@ -0,0 +1,235 @@
+        }
+
+        // Use specified policy or construct default if null.
+        cachePolicy = policy;
+        if (cachePolicy == null) {
+            cachePolicy = new CachePolicy() {
+                @Override
+                public CacheControl apply(@NonNull final HttpUrl url) {
+                    return null;
+                }
+            };
+        }
+
+        if (Build.VERSION.SDK_INT < 22) {
+            try {
+                final SSLContext sc = SSLContext.getInstance("TLSv1.2");
+                sc.init(null, null, null);
+                builder.sslSocketFactory(new Tls12SocketFactory(sc.getSocketFactory()));
+
+                final ConnectionSpec cs = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                        .tlsVersions(TlsVersion.TLS_1_2)
+                        .build();
+
+                final List<ConnectionSpec> specs = new ArrayList<>();
+                specs.add(cs);
+                specs.add(ConnectionSpec.COMPATIBLE_TLS);
+                specs.add(ConnectionSpec.CLEARTEXT);
+
+                builder.connectionSpecs(specs);
+            } catch (final Exception exc) {
+                android.util.Log.e("Tangram", "Error while setting TLS 1.2", exc);
+            }
+        }
+
+        okClient = builder.build();
+    }
+
+    @Override
+    public void onRequest(@NonNull final String url, @NonNull final HttpResponse cb, final long requestHandle) {
+        final HttpUrl httpUrl = HttpUrl.parse(url);
+        if (httpUrl == null) {
+            cb.failure(new IOException("HttpUrl failed to parse url=" + url));
+        }
+        else {
+            // Construct OkHttp.Callback which fowards apt response calls to internal Tangram.HttpResponse
+            final Callback callback = new Callback() {
+                @Override
+                public void onFailure(final Call call, final IOException e) {
+                    cb.failure(e);
+                }
+
+                @Override
+                public void onResponse(final Call call, final Response response) throws IOException {
+                    if (!response.isSuccessful()) {
+                        IOException e = new IOException("Unexpected response code: " + response + " for URL: " + url);
+                        cb.failure(e);
+                        throw e;
+                    }
+                    final ResponseBody body = response.body();
+                    if (body == null) {
+                        IOException e = new IOException("Unexpected null body for URL: " + url);
+                        cb.failure(e);
+                        throw e;
+                    }
+                    else {
+                        cb.success(body.bytes());
+                    }
+                }
+            };
+            final Request.Builder builder = new Request.Builder().url(httpUrl).tag(requestHandle);
+            final CacheControl cacheControl = cachePolicy.apply(httpUrl);
```

Comment:
Our `CachePolicy` interface also uses OkHTTP types. If we want to remove OkHTTP from our interfaces then it will also need to be refactored.